daemon/src/db/failed.rs

```diff
@@ -0,0 +1,556 @@
+//! CFDs.
+//!
+//! Failed CFDs are ones that did not reach the point of having a DLC
+//! on chain. This can happen either because the taker's order was
+//! rejected or contract setup did not finish successfully.
+//!
+//! Therefore, it also provides an interface to load failed CFDs: the
+//! `FailedCfdAggregate` trait. Implementers of the trait will be able
+//! to call the `crate::db::load_all_cfds` API, which loads all types
+//! of CFD.
+
+use crate::db;
+use crate::db::delete_from_cfds_table;
+use crate::db::delete_from_events_table;
+use crate::db::event_log::EventLog;
+use crate::db::event_log::EventLogEntry;
+use crate::db::load_cfd_events;
+use crate::db::load_cfd_row;
+use crate::db::CfdAggregate;
+use crate::db::Connection;
+use anyhow::bail;
+use anyhow::Result;
+use model::impl_sqlx_type_display_from_str;
+use model::long_and_short_leverage;
+use model::Contracts;
+use model::EventKind;
+use model::FeeAccount;
+use model::Fees;
+use model::FundingFee;
+use model::Identity;
+use model::Leverage;
+use model::OrderId;
+use model::Position;
+use model::Price;
+use model::Role;
+use model::Timestamp;
+use sqlx::pool::PoolConnection;
+use sqlx::Connection as _;
+use sqlx::Sqlite;
+use sqlx::Transaction;
+use std::fmt;
+use std::str;
+
+/// A trait for building an aggregate based on a `FailedCfd`.
+pub trait FailedCfdAggregate: CfdAggregate {
+    fn new_failed(args: Self::CtorArgs, cfd: FailedCfd) -> Self;
+}
+
+/// Data loaded from the database about a failed CFD.
+#[derive(Debug, Clone, Copy)]
+pub struct FailedCfd {
+    pub id: OrderId,
+    pub position: Position,
+    pub initial_price: Price,
+    pub taker_leverage: Leverage,
+    pub n_contracts: Contracts,
+    pub counterparty_network_identity: Identity,
+    pub role: Role,
+    pub fees: Fees,
+    pub kind: Kind,
+    pub creation_timestamp: Timestamp,
+}
+
+/// The type of failed CFD.
+#[derive(Debug, Clone, Copy)]
+pub enum Kind {
+    OfferRejected,
+    ContractSetupFailed,
+}
+
+impl fmt::Display for Kind {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let s = match self {
+            Kind::OfferRejected => "OfferRejected",
+            Kind::ContractSetupFailed => "ContractSetupFailed",
+        };
+
+        s.fmt(f)
+    }
+}
+
+impl str::FromStr for Kind {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let kind = match s {
+            "OfferRejected" => Kind::OfferRejected,
+            "ContractSetupFailed" => Kind::ContractSetupFailed,
+            other => bail!("Not a failed CFD Kind: {other}"),
+        };
+
+        Ok(kind)
+    }
+}
+
+impl_sqlx_type_display_from_str!(Kind);
+
+impl Connection {
+    pub async fn move_to_failed_cfds(&self) -> Result<()> {
+        let ids = self.failed_cfd_ids_according_to_events().await?;
+
+        if !ids.is_empty() {
+            tracing::debug!("Moving ContractSetupFailed CFDs to failed_cfds table: {ids:?}");
```

Comment:
```suggestion
            tracing::debug!("Moving failed CFDs to failed_cfds table: {ids:?}");
```

It includes both `ContractSetupFailed` and `OfferRejected`